lib/block_view_message.go

```diff
@@ -474,6 +472,26 @@ func (bav *UtxoView) _connectPrivateMessage(
 				"_connectPrivateMessage: at least one messaging party must be present")
 		}
 
+		// Reject message if sender is muted
+		var messagingGroupKey *MessagingGroupKey
+		messagingGroupKey = NewMessagingGroupKey(NewPublicKey(txMeta.RecipientPublicKey), txn.ExtraData[RecipientMessagingGroupKeyName])
```

Comment:
At this point in the function, we know that either `existsSender` or `existsRecipient` is true (L#469) but we don't know if it's the recipient, or the sender, or both. When you're referring to for example `txn.ExtraData[RecipientMessagingGroupKeyName]`, we're assuming it's not empty, which it might actually be. Check out the validation logic for the recipient and the sender on lines L#498 and L#510. I think the mutelist requires both the recipient and sender present in ExtraData. In this case you should move your code after these two conditions and gate it with a condition of `existsSender && existsSenderName && existsRecipient && existsRecipientName`.

Reply:
agreed, done